platform-modules/import-manager/src/main/scala/org/sunbird/object/importer/ImportManager.scala

```diff
@@ -53,11 +62,35 @@ class ImportManager(config: ImportConfig) {
 
 	def validateAndGetRequest(objects: util.List[util.Map[String, AnyRef]], processId: String, invalidCodes: util.List[String], invalidStages: util.List[String], request: Request)(implicit oec: OntologyEngineContext, ec: ExecutionContext): Future[util.List[util.Map[String, AnyRef]]] = {
 		Future {
+			val appIconMap: util.Map[String, String] = new util.HashMap[String, String]()
 			objects.asScala.map(obj => {
 				val source: String = obj.getOrDefault(ImportConstants.SOURCE, "").toString
 				val stage: String = obj.getOrDefault(ImportConstants.STAGE, "").toString
 				val reqMetadata: util.Map[String, AnyRef] = obj.getOrDefault(ImportConstants.METADATA, new util.HashMap[String, AnyRef]()).asInstanceOf[util.Map[String, AnyRef]]
 				val sourceMetadata: util.Map[String, AnyRef] = getMetadata(source, request.getContext.get("objectType").asInstanceOf[String].toLowerCase())
+				val appIcon = sourceMetadata.getOrDefault("appIcon","").asInstanceOf[String]
+				if(appIcon != null && appIcon.nonEmpty)
+				{
+					try {
+						if (!appIconMap.containsKey(appIcon)) {
+							val appIconFolder = CONTENT_FOLDER + File.separator + Slug.makeSlug(sourceMetadata.getOrDefault("identifier", "").asInstanceOf[String], true) + File.separator + "assets"
+							val appIconFile = downloadAppIconFile(sourceMetadata.getOrDefault("identifier", "").asInstanceOf[String], appIcon)
+							val appIconCloudUrl = ss.uploadFile(appIconFolder, appIconFile, Option(false))(1)
+							try {
+								if(appIconFile.exists()) appIconFile.delete()
+								if (new File(appIconFolder).exists() && new File(appIconFolder).isDirectory)
+									FileUtils.deleteDirectory(new File(appIconFolder))
+							} catch {
+								case e: Exception =>
+									e.printStackTrace()
+							}
+							appIconMap.put(appIcon, appIconCloudUrl)
+						}
+						sourceMetadata.put("appIcon", appIconMap.get(appIcon))
+					} catch {
+						case _:Exception =>	sourceMetadata.put("appIcon", appIcon)
+					}
```

Comment:
Do we need this try, catch still?

Reply:
in case of server exception from google drive download, we want to continue with creation of process Id. Hence, added try catch.